src/vcpkg/commands.add.cpp

```diff
@@ -41,8 +46,69 @@ namespace vcpkg::Commands
             {
                 Checks::exit_with_message(VCPKG_LINE_INFO, "add port requires an active manifest file.\n");
             }
+            const std::vector<ParsedQualifiedSpecifier> specs = Util::fmap(args.command_arguments, [&](auto&& arg) {
+                ExpectedS<ParsedQualifiedSpecifier> value = parse_qualified_specifier(std::string(arg));
+                if (value)
+                {
+                    if (!value.value_or_exit(VCPKG_LINE_INFO).triplet)
+                    {
+                        return value.value_or_exit(VCPKG_LINE_INFO);
+                    }
+                    print2(Color::error,
+                           "Error: Triplet expressions are not allowed here. Use the --triplet parameter when running "
+                           "`vcpkg install`.\n");
```

Comment:
Can you use the new localized form here? At the top of the file:

```cpp
DECLARE_AND_REGISTER_MESSAGE(
    AddTripletExpressionNotAllowed,
    (),
    "",
    "Error: Triplet expressions are not allowed here. Use the --triplet parameter when running `vcpkg install`.");
```

and then here, write:

```suggestion
                    msg::println(Color::error, AddTripletExpressionNotAllowed);
```